src/Protocol/Protocol18x.cpp

```diff
@@ -1834,39 +1834,40 @@ void cProtocol180::AddReceivedData(const char * a_Data, size_t a_Size)
 		}
 		
 		// Check packet for compression:
-		UInt32 CompressedSize = 0;
+		UInt32 UncompressedSize = 0;
 		AString UncompressedData;
 		if (m_State == 3)
 		{
 			UInt32 NumBytesRead = static_cast<UInt32>(m_ReceivedData.GetReadableSpace());
-			m_ReceivedData.ReadVarInt(CompressedSize);
-			if (CompressedSize > PacketLen)
-			{
-				m_Client->Kick("Bad compression");
-				return;
-			}
-			if (CompressedSize > 0)
+			m_ReceivedData.ReadVarInt(UncompressedSize);
```

Comment:
Isn't this checked [here](https://github.com/bibo38/Cuberite/blob/decompressfix/src/Protocol/Protocol18x.cpp#L1829), since the package length is tested to be ready to read, which includes the uncompressed size?